### PR TITLE
[releases/24.x] fix bug related to daily no. series lines

### PR DESCRIPTION
**Pull request by grobyns**
This pull request backports #1243 to releases/24.x

Fixes [AB#537309](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/537309)

### CODE REVIEW

**Review by JesperSchulz — APPROVED** [review]

File: src/Business Foundation/App/NoSeries/src/Single/NoSeriesImpl.Codeunit.al [hunk 1]
```diff
@@ -163,12 +164,21 @@ codeunit 304 "No. Series - Impl."
...
+#if not CLEAN24
+#pragma warning disable AL0432
+        NoSeriesManagement.RaiseObsoleteOnNoSeriesLineFilterOnBeforeFindLast(NoSeriesLine2);
+#pragma warning restore AL0432
+#endif
+        if NoSeriesLine2.FindLast() then;
+        NoSeriesLine.CopyFilters(NoSeriesLine2);
+
+        if (NoSeriesLine."Line No." <> 0) and (NoSeriesLine."Series Code" = NoSeriesCode) and (NoSeriesLine."Starting Date" = NoSeriesLine2."Starting Date") then begin
+            NoSeriesLine.CopyFilters(NoSeriesLine2);
```
Comment (AndreasMoth) [inline src/Business Foundation/App/NoSeries/src/Single/NoSeriesImpl.Codeunit.al:181]: This is mainly for main branch (but PR closed), I see we are copying filters here again. A bit unnecessary as we already copied them above :)

File: src/Business Foundation/App/NoSeries/src/Single/NoSeriesImpl.Codeunit.al [hunk 2]
```diff
@@ -163,12 +164,21 @@ codeunit 304 "No. Series - Impl."
             UsageDate := WorkDate();
 
         // Find the No. Series Line closest to the usage date
-        NoSeriesLine.Reset();
-        NoSeriesLine.SetCurrentKey("Series Code", "Starting Date");
-        NoSeriesLine.SetRange("Series Code", NoSeriesCode);
-        NoSeriesLine.SetRange("Starting Date", 0D, UsageDate);
-        NoSeriesLine.SetRange(Open, true);
-        if (NoSeriesLine."Line No." <> 0) and (NoSeriesLine."Series Code" = NoSeriesCode) then begin
+        NoSeriesLine2.Reset();
```
Comment (AndreasMoth) [inline src/Business Foundation/App/NoSeries/src/Single/NoSeriesImpl.Codeunit.al:167]: Minor: No need to reset NoSeriesLine2

File: src/Business Foundation/App/NoSeries/src/Single/NoSeriesImpl.Codeunit.al (hunk 1 above)
Comment (grobyns) [reply src/Business Foundation/App/NoSeries/src/Single/NoSeriesImpl.Codeunit.al:181]: will fix next time...

File: src/Business Foundation/App/NoSeries/src/Single/NoSeriesImpl.Codeunit.al (hunk 2 above)
Comment (grobyns) [reply src/Business Foundation/App/NoSeries/src/Single/NoSeriesImpl.Codeunit.al:167]: will fix next time...

**Review by nikolakukrika — APPROVED** [review]

**Pull request merged**